### PR TITLE
Reduce usage of `Lazy` for ReadOnly mirror of a collection

### PR DESCRIPTION
**Pull request by DubyaDude**
# Summary
Lazy doesn't seem like a proper solution for the ReadOnly, and it may be a culprit for some high memory usage.

# Details
- ~~TBD (will fill out with actual changes in memory usage later)~~

# Changes proposed
I propose just returning the collection as an `IReadOnly`, while this doesn't stop the user from casting it back, I feel just returning it as IReadOnly should be enough.

### CODE REVIEW

**Comment by Plerx2493** [comment]
I personally would prefer returning real RO-Collections/Dicts. I think thats also the case in most other places we expose things like that

**Review by akiraveliara — APPROVED** [review]
lgtm, even if this doesn't bring a marked improvement in performance reducing Lazy cruft is always worthwhile

**Review by Plerx2493 — APPROVED** [review]

**Comment by DubyaDude** [comment]
Since maintainers would like this change aside from any potential memory gains, will go forward without analyzing it.

**Pull request merged**
